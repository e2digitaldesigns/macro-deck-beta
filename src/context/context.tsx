import React from "react";
import { ThemeContextProvider } from "./themeContext";

export interface IThemeComponents {
  ThemeProvider: React.ElementType;
}

export const ApplicationContext: React.FC & IThemeComponents = () => <div />;

ApplicationContext.ThemeProvider = ThemeContextProvider;
