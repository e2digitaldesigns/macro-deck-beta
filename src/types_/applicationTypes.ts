export type Profile = {
  _id: string;
  order: number;
  buttonPads: number;
  profileName: string;
};

export type Page = {
  _id: string;
  profileId: string;
  order: number;
};

export type ButtonPad = {
  _id: string;
  bgColor: string;
  buttonPadNum: number;
  icon: string;
  iconColor: string;
  image: string;
  pageId: string;
  profileId: string;
  text: string;
  textColor: string;
};

export type ActionData = {
  plugin: string;
  pluginAction: string;
  displayText: string;
  pluginData: object;
};

export type Action = {
  _id: string;
  action: string;
  buttonPadId: string;
  data: ActionData;
  layer: string;
  order: number;
  page: string;
  pageId: string;
  path: string;
  profile: string;
  profileId: string;
  scene: string;
  seconds: number;
  subAction: string;
  text: string;
  url: string;
  volume: number;
};

export type Style = {
  _id: string;
  bgColor: string;
  icon: string;
  iconColor: string;
  image: string;
  textColor: string;
};

export enum ButtonPadNums {
  bpn06 = 6,
  bpn08 = 8,
  bpn12 = 12,
  bpn15 = 15,
  bpn24 = 24,
  bpn32 = 32
}

export enum ButtonPadGridSize {
  BPGS32 = 32,
  BPGS36 = 36
}

export const AdjustGridButtonPadNums: ButtonPadNums[] = [
  ButtonPadNums.bpn06,
  ButtonPadNums.bpn15
];

export enum SectionRoutes {
  Home = "home",
  MacroDeck = "macroDeck",
  Settings = "settings",
  SettingsMacroDeck = "settings/macroDeck",
  SettingsObs = "settings/obs",
  SettingsServer = "settings/server",
  SettingsTwitch = "settings/twitch",
  Splash = "/splash"
}
