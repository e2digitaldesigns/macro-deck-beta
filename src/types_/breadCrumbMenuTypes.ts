export enum BreadCrumbMenuTypes {
  Profile = "profile",
  Page = "page",
  ButtonPad = "buttonPad",
  Action = "action"
}
