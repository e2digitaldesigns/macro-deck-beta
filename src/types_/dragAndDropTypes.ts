export enum DragAndDropOptions {
  CopyButtonPad = "CopyButtonPad",
  MoveButtonPad = "MoveButtonPad",
  OverwriteButtonPad = "OverwriteButtonPad",
  PasteButtonPad = "PasteButtonPad",
  StyleButtonPad = "StyleButtonPad",
  SwapButtonPad = "SwapButtonPad"
}

export enum DragAndDropDataTypes {
  Action = "DragAndDropAction",
  PageId = "DragAndDropPageId",
  StyleId = "DragAndDropStyleId",
  OriginPadNumber = "DragAndDropOriginPadNumber",
  OriginPadId = "DragAndDropOriginPadId"
}
