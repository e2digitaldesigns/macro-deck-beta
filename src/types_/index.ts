export * from "./breadCrumbMenuTypes";
export * from "./applicationTypes";
export * from "./profileModes";
export * from "./ipcRenderTypes";
export * from "./dragAndDropTypes";
export * from "./ipcRenderTypes";
export * from "./subActionTypes";

export interface ISettingsFeaturesObs {
  ipAddress: string;
  port: string;
  password: string;
  status?: boolean;
}

export interface IMDSettings {
  ipAddress: string;
  port: string;
}

export enum FormFieldFileTypes {
  MDFileFieldExe = "mdFileFieldExe",
  MDFileFieldSound = "mdFileFieldSound",
  FileField = "fileField"
}
