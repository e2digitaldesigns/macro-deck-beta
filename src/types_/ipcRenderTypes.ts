export enum IpcRendererTypes {
  database = "database",
  databaseMobileUpdate = "database:mobileUpdate",
  header = "header",
  mdPlaySound = "MacroDeck:playSound",
  Ping = "ping",
  soundBoardProfile = "soundBoardProfile:dialogueBox",
  soundBoardProfileReturn = "soundBoardProfile:dialogueBoxReturn",
  FetchMDSettings = "fetchMDSettings",
  FetchMDSettingsReturn = "fetchMDSettings:return"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData",
  updateMobileDevice = "updateMobileDevice"
}
