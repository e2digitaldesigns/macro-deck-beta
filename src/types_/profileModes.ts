export enum ProfileModes {
  Edit = "edit",
  New = "new",
  View = "view",
  NewSoundBoard = "newSoundBoard"
}

export enum profileDefaultsOptions {
  ProfileEmpty = "profileEmpty",
  ProfileMD = "profileMD",
  ProfileStream = "profileStream",
  ProfileMediaControls = "profileMediaControls",
  ProfileMusic = "profileMusic",
  ProfileSoundBoard = "profileSoundBoard"
}
