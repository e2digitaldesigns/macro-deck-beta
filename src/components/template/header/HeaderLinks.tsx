import React from "react";
import { Link, useLocation } from "react-router-dom";
import { Icon } from "react-bootstrap-icons";
import * as Styled from "./TemplateHeader.style";
import { SectionRoutes } from "../../../types_";

export interface IntHeaderLinks {
  Icon1: Icon;
  Icon2: Icon;
  name: string;
  route: SectionRoutes;
}

export const HeaderLinks: React.FC<IntHeaderLinks> = ({
  Icon1,
  Icon2,
  name,
  route
}) => {
  const location = useLocation();
  const path = location.pathname.split("/");
  const isActive = route === path[1];

  return (
    <Styled.LinkWrapper isActive={isActive} data-testid="header-link">
      {isActive ? (
        <Icon2 data-testid="header-link__active-icon" />
      ) : (
        <Icon1 data-testid="header-link__normal-icon" />
      )}
      <Link data-testid="header-link__name" to={route}>
        {name}
      </Link>
    </Styled.LinkWrapper>
  );
};
