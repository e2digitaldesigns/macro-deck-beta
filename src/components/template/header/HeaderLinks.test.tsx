import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { HashRouter as Router } from "react-router-dom";
import * as rrd from "react-router-dom";

import { HeaderLinks, IntHeaderLinks } from "./HeaderLinks";
import { SectionRoutes } from "../../../types_";

import { HouseDoor, HouseDoorFill } from "react-bootstrap-icons";
import { ApplicationContext } from "../../../context/context";

const defaultProps: IntHeaderLinks = {
  Icon1: HouseDoor,
  Icon2: HouseDoorFill,
  name: "Home",
  route: SectionRoutes.Home
};

const mockConfig = rrd as { useLocation: () => { pathname: string } };

jest.mock("react-router-dom", () => ({
  __esModule: true,
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/home"
  })
}));

const setup = (props: IntHeaderLinks = defaultProps) => {
  return render(
    <ApplicationContext.ThemeProvider>
      <Router>
        <HeaderLinks {...defaultProps} />{" "}
      </Router>
    </ApplicationContext.ThemeProvider>
  );
};

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe("HeaderLinks", () => {
  it("Should render without errors", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });

  it("should have correct text", () => {
    const component = setup();
    expect(component.getByTestId("header-link__name")).toBeTruthy();
  });

  it("should have correct icon (active)", () => {
    const component = setup();
    expect(component.getByTestId("header-link__active-icon")).toBeTruthy();
  });

  it("should have correct icon (normal)", () => {
    mockConfig.useLocation = () => ({
      pathname: "/settings"
    });
    const component = setup();

    expect(component.getByTestId("header-link__normal-icon")).toBeTruthy();
  });
});
