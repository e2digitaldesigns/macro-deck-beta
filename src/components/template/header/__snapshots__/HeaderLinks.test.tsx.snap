// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLinks Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsnTZi jomnHy"
        data-testid="header-link"
      >
        <svg
          data-testid="header-link__active-icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.5 14.5v-3.505c0-.245.25-.495.5-.495h2c.25 0 .5.25.5.5v3.5a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 .5-.5v-7a.5.5 0 0 0-.146-.354L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293L8.354 1.146a.5.5 0 0 0-.708 0l-6 6A.5.5 0 0 0 1.5 7.5v7a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 .5-.5z"
          />
        </svg>
        <a
          data-testid="header-link__name"
          href="#/home"
        >
          Home
        </a>
      </div>
       
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsnTZi jomnHy"
      data-testid="header-link"
    >
      <svg
        data-testid="header-link__active-icon"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.5 14.5v-3.505c0-.245.25-.495.5-.495h2c.25 0 .5.25.5.5v3.5a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 .5-.5v-7a.5.5 0 0 0-.146-.354L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293L8.354 1.146a.5.5 0 0 0-.708 0l-6 6A.5.5 0 0 0 1.5 7.5v7a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 .5-.5z"
        />
      </svg>
      <a
        data-testid="header-link__name"
        href="#/home"
      >
        Home
      </a>
    </div>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
